src/essence/Tools/Isochrone/IsochroneTool.js

```diff
@@ -0,0 +1,539 @@
+        let canvas = document.createElement("canvas");
+        canvas.width = 16;
+        canvas.height = 16;
+        let ctx = canvas.getContext("2d");
+
+        const radius = 7
+        const strokeWeight = 2
+        const ramp = IsochroneTool.colorRamps[options.color];
+        const c = ramp[ramp.length - 1];
+
+        ctx.fillStyle = `rgba(${c[0]}, ${c[1]}, ${c[2]}, 255)`;
+
+        ctx.strokeStyle = "rgba(255, 255, 255, 255)";
+        ctx.beginPath();
+        ctx.arc(
+            canvas.width / 2,
+            canvas.height / 2,
+            radius,
+            0,
+            2 * Math.PI
+        );
+
+        ctx.fill();
+        ctx.lineWidth = strokeWeight;
+        ctx.stroke();
+        ctx.strokeStyle = "rgba(0, 0, 0, 255)";
+        ctx.beginPath();
+        ctx.arc(
+            canvas.width / 2,
+            canvas.height / 2,
+            radius - strokeWeight,
+            0,
+            2 * Math.PI
+        );
+
+        ctx.fill();
+        ctx.lineWidth = strokeWeight;
+        ctx.stroke();
+
+        let isochroneIcon = L.icon({
+            iconUrl: canvas.toDataURL(),
+            iconSize: [canvas.width, canvas.height],
+            iconAnchor: [canvas.width / 2, canvas.height / 2],
+            popupAnchor: [-3, -76],
+            shadowSize: [68, 95],
+            shadowAnchor: [22, 94],
+        });
+
+        Map_.rmNotNull(manager.marker);
+        manager.marker = new L.marker([start.lat, start.lng], {
+            icon: isochroneIcon,
+            draggable
+        });
+
+        manager.marker.on("click", e => this.managerOnFocus(manager));
+        if(draggable) {
+            manager.marker.on("dragend", e => {
+                manager.start = e.target._latlng;
+                manager.setBounds();
+                this.managerOnFocus(manager);
+            });
+        }
+        manager.marker.addTo(Map_.map);
+    },
+
+    handleClick: function(e) {
+        if(e && e.latlng) {
+            this.activeManager.start = e.latlng;
+            this.makeMarker(this.activeManager, false);
+            this.activeManager.setBounds();
+        }
+    },
+
+    handleMove: function(e) {
+        const MAX_STEPS = 5000;
+        const manager = this.activeManager;
+
+        if(manager.backlink === null || manager.currentStage > 0 || !manager.options.visible)
+            return;
+        
+        const now = Date.now();
+        if(this.lastHoverCall + 65 > now) return;
+        this.lastHoverCall = now;
+
+        if(this.hoverPolyline !== null)
+            this.hoverPolyline.remove(Map_.map);
+
+        const toLinePoint = (x, y) => {
+            const point = manager.tileBounds.min
+                .multiplyBy(256)
+                .add([x + 0.5, y + 0.5]);
+            const latlng = Map_.map.unproject(point, manager.options.resolution);
+            return [latlng.lat, latlng.lng];
+        }
+
+        const hoveredPx = Map_.map
+            .project(e.latlng, manager.options.resolution)
```

Comment:
when using the tool without any data layers, I'm seeing "`e` is null" errors for every mouse move in the console. I believe they're coming from here. you should check that `e` exists and is what you expect it to be at the top of this function. or, you should avoid calling this function if there is no `e`

Reply:
I'm seeing this even with data in the [JSC Rock Yard project](https://mmgis.aresxi.space/?mission=JSC_RockYard), which I cloned locally

Reply:
<img width="1625" alt="Screen Shot 2021-08-06 at 10 47 32" src="https://user-images.githubusercontent.com/5186335/128537043-a762cb46-a6a1-42b9-a030-123c20b96266.png">